facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java

```diff
@@ -603,49 +597,82 @@ private org.jboss.pnc.model.Artifact.Builder mapArtifact(Artifact artifact, User
     }
 
-    private String createBrewOriginURL(MavenArtifact mavenArt, String nvr) {
+    private String createBrewOriginURL(Artifact artifact, String nvr) {
         String brewContentUrl = globalConfig.getBrewContentUrl();
-
         Matcher matcher = NVR_PATTERN.matcher(nvr);
         if (!matcher.matches()) {
             throw new IllegalArgumentException("NVR " + nvr + " does not match expected format.");
         }
         String name = matcher.group(1);
         String version = matcher.group(2);
         String release = matcher.group(3);
-
-        return brewContentUrl + "/" + name + "/" + version + "/" + release + "/maven" + createDeployPath(mavenArt);
+        return brewContentUrl + "/" + name + "/" + version + "/" + release + "/"
+                + artifact.getArtifactType().name().toLowerCase(Locale.ENGLISH) + createDeployPath(artifact);
     }
 
-    private String createPURL(MavenArtifact mavenArtifact) {
+    private String createPURL(Artifact artifact) {
         try {
-            PackageURLBuilder purlBuilder = PackageURLBuilder.aPackageURL()
-                    .withType(PackageURL.StandardTypes.MAVEN)
-                    .withNamespace(mavenArtifact.getGroupId())
-                    .withName(mavenArtifact.getArtifactId())
-                    .withVersion(mavenArtifact.getVersion())
-                    .withQualifier(
-                            "type",
-                            StringUtils.isEmpty(mavenArtifact.getType()) ? "jar" : mavenArtifact.getType());
-
-            if (!StringUtils.isEmpty(mavenArtifact.getClassifier())) {
-                purlBuilder.withQualifier("classifier", mavenArtifact.getClassifier());
+            PackageURLBuilder purlBuilder;
+            if (artifact instanceof MavenArtifact) {
+                MavenArtifact mavenArtifact = (MavenArtifact) artifact;
+                purlBuilder = PackageURLBuilder.aPackageURL()
+                        .withType(PackageURL.StandardTypes.MAVEN)
+                        .withNamespace(mavenArtifact.getGroupId())
+                        .withName(mavenArtifact.getArtifactId())
+                        .withVersion(mavenArtifact.getVersion())
+                        .withQualifier(
+                                "type",
+                                StringUtils.isEmpty(mavenArtifact.getType()) ? "jar" : mavenArtifact.getType());
+
+                if (!StringUtils.isEmpty(mavenArtifact.getClassifier())) {
+                    purlBuilder.withQualifier("classifier", mavenArtifact.getClassifier());
+                }
+
+            } else if (artifact instanceof WindowsArtifact) {
+                WindowsArtifact windowsArtifact = (WindowsArtifact) artifact;
+                purlBuilder = PackageURLBuilder.aPackageURL()
+                        .withType(PackageURL.StandardTypes.GENERIC)
+                        .withName(windowsArtifact.getName())
+                        .withVersion(windowsArtifact.getVersion());
```

Comment:
Namespace?

Reply:
@pkocandr Should the namespace be empty? Type is `GENERIC` as there is no `WINDOWS` type.